designdocs/wallettool-subcommands.adoc

```diff
@@ -0,0 +1,579 @@
+----
+
+=== *`encrypt`*
+
+*Description*: Encrypts the wallet using the specified password. Requires `--password`.
+
+==== *Usage*:
+
+[source,bash]
+----
+wallet-tool <WALLET_FILE> encrypt --password <PASSWORD>
+----
+
+==== *Options*:
+* `--password <PASSWORD>`: Password to encrypt the wallet.
+
+==== *Examples*:
+
+[arabic]
+. Encrypt the wallet with a password:
++
+[source,bash]
+----
+wallet-tool example.wallet encrypt --password strongpassword123
+----
+
+=== *`decrypt`*
+
+*Description*: Decrypts the wallet using the provided password. Requires
+`--password`
+
+==== *Usage*:
+
+[source,bash]
+----
+wallet-tool decrypt --wallet <WALLET_FILE> --password <PASSWORD>
+----
+
+==== *Options*:
+
+* `--wallet ++<++WALLET++_++FILE++>++`: The wallet file to decrypt.
+* `--password ++<++PASSWORD++>++`: Password used to decrypt the wallet.
+
+==== *Examples*:
+
+[arabic]
+. Decrypt an encrypted wallet:
++
+[source,bash]
+----
+wallet-tool decrypt --wallet example.wallet --password strongpassword123
+----
+
+=== *`send`*
+
+*Description*: Creates and broadcasts a transaction from the given wallet. Requires `--output` to be specified.
+
+==== *Usage*:
+
+[source,bash]
+----
+wallet-tool <WALLET_FILE> send --output <ADDRESS>:<AMOUNT> [--password <PASSWORD>] [--fee-per-vkb <FEE>] [--fee-sat-per-vbyte <FEE>] [--locktime <BLOCK/TIME>] [--select-addr <ADDRESS>] [--select-output <OUTPUT>] [--waitfor <CONFIRMATIONS>] [--allow-unconfirmed] [--no-pki]
+----
+
+==== *Options*:
+
+* `--output ++<++ADDRESS++>++:++<++AMOUNT++>++`: Target address and amount. If specified, a transaction is created from the provided output from this wallet and broadcast. (e.g., `1GthXFQMktFLWdh5EPNGqbq3H6WdG8zsWj:1.245`). You can repeat `--output=address:value` multiple times. There is a magic value `ALL` which empties the wallet to that address, e.g., `--output=1GthXFQMktFLWdh5EPNGqbq3H6WdG8zsWj:ALL`. The output destination can also be a native segwit address. If the output destination starts with 04 and is 65 or 33 bytes long it will be treated as a public key instead of an address and the `send` will use `<key> CHECKSIG` as the script.
+* `--password ++<++PASSWORD++>++`: Password to unlock the wallet (if encrypted).
+* `--fee-per-vkb ++<++FEE++>++`: Sets the network fee. Transaction fee per kilobyte.
+* `--fee-sat-per-vbyte ++<++FEE++>++`: Sets the network fee. Transaction fee per virtual byte in satoshis.
+* `--locktime ++<++BLOCK/TIME++>++`: Specifies a lock-time either by date (e.g., `--locktime=2013/01/01`) or by block number (e.g., `--locktime=1234`).
+* `--select-addr ++<++ADDRESS++>++`: Restricts input selection to a specific address.
+* `--select-output ++<++OUTPUT++>++`: Restricts input selection to a specific output.
+* `--waitfor ++<++CONFIRMATIONS++>++`: Waits for a specific number of confirmations.
+* `--allow-unconfirmed`: Allows you to create spends of pending non-change outputs.
+* `--no-pki`: Disables PKI verification for payment requests.
```

Comment:
Payment requests have been removed (I think, if not will be removed soon) so we should remove this option.